src/main.c

```diff
@@ -184,11 +200,16 @@ static inline void *romdec_dmaext(unsigned char *rom, size_t romSz, size_t *dstS
 		return NULL;
 	}
 
+	/* since we now know where the end of dmadata is, we can allocate the list of
+		compressed and uncompressed files for printing the z64compress args later. */
+	/* Add one for the terminator */
+	fileIsCompressed = calloc(1, sizeof(signed char) * (((dmaEnd - dmaStart) / 4) + 1));
+
 	/* allocate decompressed rom */
 	dec = calloc_safe(*dstSz, 1);
 
 	/* transfer files from comp to dec, and decompress them if needed */
-	for (dmaCur = dmaStart; dmaCur < dmaEnd; ) 
+	for (dmaNum = 0, dmaCur = dmaStart; dmaCur < dmaEnd; dmaNum++) 
```

Comment:
Todo: there is an inconsistency in naming with this between the other decompress function